packages/eslint-plugin/src/rules/consistent-type-definitions.ts

```diff
@@ -68,7 +68,9 @@ export default util.createRule({
           });
         }
       },
-      TSInterfaceDeclaration(node): void {
+      ':not(TSModuleDeclaration > TSModuleBlock) > TSInterfaceDeclaration'(
+        node: TSESTree.TSInterfaceDeclaration,
+      ): void {
```

Comment:
this will stop the rule reporting at all when an interface is within a module declaration.
Which is too wide of a change.

Instead, we still want to report on interface within a module declaration - but we want to just no automatically fix it if and only if it's within a `declare global`

Reply:
What's the manual fix for that case?

Reply:
the manual fix in that case would be for the engineer to review the lint and disable it in that case.

it's impossible for us to tell exactly what the intentions are so we have three options:
- report with a fixer
  - this is bad because it means we can break the user's code.
- do nothing
  - this isn't great because it means that the user can easily create code which should be marked as invalid.
  - it's impossible to tell if the code was written invalidly because it HAD to be this way, or if the author just wrote it wrong.
- report without a fixer
  - this is a best-of-both worlds.
  - if the code is actually invalid - the user can fix it
  - if it's an edge case - the user can suppress the lint rule to show it's intentionally written this way.

> I thought it's impossible to use the type within a declare in the case so we have to turn off it completely

Nope - it's perfectly acceptable to use `type` declarations or `interface` declarations.
In some cases the latter is better as it will allow specifically for augmentation via declaration merging.